presto-hive-functions/src/main/java/com/facebook/presto/hive/functions/HiveFunctionNamespaceManager.java

```diff
@@ -0,0 +1,284 @@
+    }
+
+    @Override
+    public Collection<HiveFunction> listFunctions(Optional<String> likePattern, Optional<String> escape)
+    {
+        // TODO Implement list functions
+        return Collections.emptyList();
+    }
+
+    public Collection<HiveFunction> getFunctions(Optional<? extends FunctionNamespaceTransactionHandle> transactionHandle, QualifiedObjectName functionName)
+    {
+        throw new PrestoException(NOT_SUPPORTED, "Not supported");
+    }
+
+    public FunctionHandle getFunctionHandle(Optional<? extends FunctionNamespaceTransactionHandle> transactionHandle, Signature signature)
+    {
+        return new HiveFunctionHandle(requireNonNull(signature, "signature is null"));
+    }
+
+    public FunctionHandle resolveFunction(Optional<? extends FunctionNamespaceTransactionHandle> transactionHandle, QualifiedObjectName functionName, List<TypeSignature> parameterTypes)
+    {
+        return new HiveFunctionHandle(createFunction(FunctionKey.of(functionName, parameterTypes)).getSignature());
+    }
+
+    public FunctionMetadata getFunctionMetadata(FunctionHandle functionHandle)
+    {
+        checkArgument(functionHandle instanceof HiveFunctionHandle);
+        HiveFunction function = functions.getUnchecked(FunctionKey.from((HiveFunctionHandle) functionHandle));
+        return function.getFunctionMetadata();
+    }
+
+    public ScalarFunctionImplementation getScalarFunctionImplementation(FunctionHandle functionHandle)
+    {
+        checkArgument(functionHandle instanceof HiveFunctionHandle);
+        HiveFunction function = functions.getUnchecked(FunctionKey.from((HiveFunctionHandle) functionHandle));
+        verify(function instanceof HiveScalarFunction);
+        return ((HiveScalarFunction) function).getJavaScalarFunctionImplementation();
+    }
+
+    @Override
+    public CompletableFuture<SqlFunctionResult> executeFunction(String source, FunctionHandle functionHandle, Page input, List<Integer> channels, TypeManager typeManager)
+    {
+        return null;
+    }
+
+    @Override
+    public void addUserDefinedType(UserDefinedType userDefinedType)
+    {
+        return;
+    }
+
+    @Override
+    public Optional<UserDefinedType> getUserDefinedType(QualifiedObjectName typeName)
+    {
+        return Optional.empty();
+    }
+
+    @Override
+    public boolean canResolveFunction()
+    {
+        return true;
+    }
+
+    private HiveFunction createFunction(FunctionKey key)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
+            QualifiedObjectName name = key.getName();
+            try {
+                Class<?> functionClass = hiveFunctionRegistry.getClass(name);
+                if (anyAssignableFrom(functionClass, GenericUDF.class, UDF.class)) {
+                    return HiveScalarFunction.create(functionClass, name, key.getArgumentTypes(), typeManager);
```

Comment:
We generally name the static function more explicitly and use static import. So something like `createHiveScalarFunction`.